java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/InvocationContext.java

```diff
@@ -28,7 +29,13 @@ public InvocationContext(
         this.toolCallBehavior = unmodifiableClone(toolCallBehavior);
     }
 
-    @SuppressFBWarnings(value="EI_EXPOSE_REP", justification="returns UnmodifiableKernelHooks")
```

Comment:
What's the downside to removing InvocationContext altogether?

Reply:
you will end up passing a lot of extra arguments around in many locations